vizro-ai/src/vizro_ai/_vizro_ai.py

```diff
@@ -97,7 +104,7 @@ def _get_chart_code(self, df: pd.DataFrame, user_input: str) -> str:
 
     def plot(
         self, df: pd.DataFrame, user_input: str, explain: bool = False, max_debug_retry: int = 3
-    ) -> Union[None, Dict[str, Any]]:
+    ) -> Union[go.Figure, Dict[str, Any]]:
```

Comment:
Is that true? I think it is different:

This ultimately returns the following fn:

```python
def _exec_code_and_retrieve_fig(
    code: str, local_args: Optional[Dict] = None, is_notebook_env: bool = True
) -> go.Figure:
```
so it should just return a `go.Figure` I think.

Could we please be more careful in our typing here and additionally check why `mypy` did not flag this.

Reply:
I think we have a Union of go.Figure and Dict[str, Any] because of this:
```
if self._return_all_text:
    return output_dict
```

Reply:
ah yes, this one we enabled before for Alexey's integration test, later on if we aligned on plot call to return customized output as dictionary, then the type will be the same

Reply:
i would still prefer `_exec_code_and_retrieve_fig` since it returns a fig object in the end, not usually what execute code do, but it is minor i would not argue if others hold strong opinion on it. `_exec_fig_code_display_markdown` is good.

For all the others, i thought the main reason of giving output dictiionary is to give public options for user to retrieve any outputs instead of using hidden private ways.

Reply:
Let's close this for now and pick it up later. Discussed with @Anna-Xiong